core/src/main/java/io/seata/core/context/RootContext.java

```diff
@@ -46,7 +49,26 @@
      * @return the xid
      */
     public static String getXID() {
-        return CONTEXT_HOLDER.get(KEY_XID);
+        String xid = CONTEXT_HOLDER.get(KEY_XID);
+        if (StringUtils.isNotBlank(xid)) {
+            return xid;
+        }
+
+        String xidType = CONTEXT_HOLDER.get(KEY_XID_INTERCEPTOR_TYPE);
```

Comment:
If using datasourceProxy how to make it not repeat the registration branch

Reply:
Is my fault, in the connection to determine whether it is used in the transaction inGlobalTransaction.